stock_warehouse_calendar_orderpoint/models/stock_warehouse.py

```diff
@@ -56,3 +58,29 @@ def create(self, vals):
         if vals.get("company_id"):
             self = self.with_context(force_wh_company=vals["company_id"])
         return super().create(vals)
+
+    def _get_lead_date(self, date_order, lead_days):
+        self.ensure_one()
+        # Get the WH calendar
+        calendar = self.calendar_id
+        if not calendar:
+            # No WH calendar defined => consume ``lead_days`` as solar days
+            return date_order + relativedelta(days=lead_days or 0)
```

Comment:
We could avoid the import of `dateutil.relativedelta.relativedelta` and use Odoo fields API instead:
```python
return fields.Datetime.add(date_order, days=lead_days or 0)
```